src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -0,0 +1,306 @@
+    this.onQueryStringChange(value);
+  };
+
+  private onClickInput = (event: React.MouseEvent<HTMLTextAreaElement>) => {
+    if (event.target instanceof HTMLTextAreaElement) {
+      this.onQueryStringChange(event.target.value);
+    }
+  };
+
+  // TODO: MQL consider moving language select language of setting search source here
+  private onSelectLanguage = (language: string) => {
+    // Send telemetry info every time the user opts in or out of kuery
+    // As a result it is important this function only ever gets called in the
+    // UI component's change handler.
+    this.services.http.post('/api/opensearch-dashboards/dql_opt_in_stats', {
+      body: JSON.stringify({ opt_in: language === 'kuery' }),
+    });
+
+    const newQuery = {
+      query: this.props.getQueryStringInitialValue?.(language) ?? '',
+      language,
+    };
+
+    const fields = this.props.settings.getQueryEnhancements(newQuery.language)?.fields;
+    const newSettings: DataSettings = {
+      userQueryLanguage: newQuery.language,
+      userQueryString: newQuery.query,
+      ...(fields && { uiOverrides: { fields } }),
+    };
+    this.props.settings?.updateSettings(newSettings);
+
+    const dateRangeEnhancement = this.props.settings.getQueryEnhancements(language)?.searchBar
+      ?.dateRange;
+    const dateRange = dateRangeEnhancement
+      ? {
+          from: dateRangeEnhancement.initialFrom!,
+          to: dateRangeEnhancement.initialTo!,
+        }
+      : undefined;
+    this.onChange(newQuery, dateRange);
+    this.onSubmit(newQuery, dateRange);
+  };
+
+  private initPersistedLog = () => {
+    const { uiSettings, storage, appName } = this.services;
+    this.persistedLog = this.props.persistedLog
+      ? this.props.persistedLog
+      : getQueryLog(uiSettings, storage, appName, this.props.query.language);
+  };
+
+  public onMouseEnterSuggestion = (index: number) => {
+    this.setState({ index });
+  };
+
+  textareaId = htmlIdGenerator()();
+
+  public componentDidMount() {
+    const parsedQuery = fromUser(toUser(this.props.query.query));
+    if (!isEqual(this.props.query.query, parsedQuery)) {
+      this.onChange({ ...this.props.query, query: parsedQuery });
+    }
+
+    this.initPersistedLog();
+    // this.fetchIndexPatterns().then(this.updateSuggestions);
+    this.handleListUpdate();
+
+    window.addEventListener('scroll', this.handleListUpdate, {
+      passive: true, // for better performance as we won't call preventDefault
+      capture: true, // scroll events don't bubble, they must be captured instead
+    });
+  }
+
+  public componentDidUpdate(prevProps: Props) {
+    const parsedQuery = fromUser(toUser(this.props.query.query));
+    if (!isEqual(this.props.query.query, parsedQuery)) {
+      this.onChange({ ...this.props.query, query: parsedQuery });
+    }
+
+    this.initPersistedLog();
+  }
+
+  public componentWillUnmount() {
+    if (this.abortController) this.abortController.abort();
```

Comment:
nit: is code style not to use braces on single line ifs? personally think they should be required (e.g. apple's tls bug)

Reply:
this part was pretty much ported over.

i agree. one of things that we forked was the ability for plugins that have unrestricted control.

our repo has linters defined here https://github.com/opensearch-project/OpenSearch-Dashboards/tree/main/packages/opensearch-eslint-config-opensearch-dashboards

which we can add this rule (and more rules) but in the past worried about updating this. any plugin that uses our rules then will also have to update their code. And we tried our best to avoid breaking them by opening an issue in each repo and letting them know ahead of time.

i've added a line item for code clean up in both components:
ttps://github.com/opensearch-project/OpenSearch-Dashboards/issues/6957